### PR TITLE
add form field masking

### PR DESCRIPTION
**Pull request by brendansudol**
### This pull request...
- adds this nifty library (https://text-mask.github.io/text-mask/) so we can have nicely formatted inputs as users type stuff
- adds dollar and percent `MaskedInput`s to the appropriate activity form fields 

Addresses: https://github.com/18F/cms-hitech-apd/issues/515

Preview:
https://cl.ly/rdPp

### This pull request is ready to merge when...
- [ ] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author

### This feature is done when...
- [ ] Design has approved the experience

### CODE REVIEW

File: web/src/containers/ActivityDetailContractorResources.js [hunk 1]
```diff
@@ -26,7 +27,7 @@ class ActivityDetailContractorExpenses extends Component {
     const { activity, updateActivity } = this.props;
 
     const updates = {
-      contractorResources: { [index]: { years: { [year]: +value } } }
+      contractorResources: { [index]: { years: { [year]: value } } }
```
Comment (mgwalker) [inline web/src/containers/ActivityDetailContractorResources.js]: I'm not wild about this.  The text mask input sends a formatted string (e.g., `$1,000.00`) with the `onChange` event.  That's problematic for two reasons:

1. if we add that formatted string to our internal state, we can't easily use it for math later on
2. we want to store numeric data on the API for the same reason

This change will actually break #544 _right now_, so it's not a future worry.  Would it be sensible to further wrap the text mask with specific `<DollarInput>` and `<PercentInput>` components that de-format the text and send numbers in their `onChange` events?

Comment (brendansudol) [reply web/src/containers/ActivityDetailContractorResources.js]: yes, i hear you and agree -- i feel like this library should really handle this (returning the raw and formatted values) -- maybe it does, i'll check!

Comment (brendansudol) [reply web/src/containers/ActivityDetailContractorResources.js]: alright, see what you think!

File: web/src/components/Inputs.js [hunk 2]
```diff
@@ -57,6 +59,44 @@ const Textarea = props => (
   <textarea className="m0 textarea" spellCheck="true" {...props} />
 );
 
+const makeMaskedInput = ({ mask, unmask }) => {
+  // TODO: split some of these components into their own file
+  // eslint-disable-next-line react/no-multi-comp
```
Comment (mgwalker) [inline web/src/components/Inputs.js]: 👍

File: web/src/util/masks.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import createNumberMask from 'text-mask-addons/dist/createNumberMask';
+
+const toNum = x => (x.length ? +x.replace(/\D+/g, '') : '');
```
Comment (mgwalker) [inline web/src/util/masks.js]: Only concern is I think this will convert `$123.45` to `12345`.

**Pull request merged**